tools/sctool/src/main/java/com/horizen/ScBootstrappingToolCommandProcessor.java

```diff
@@ -343,6 +346,130 @@ private void processGenerateCertProofInfo(JsonNode json) {
         printer.print(res);
     }
 
+    private void processGenerateCertWithKeyRotationProofInfo(JsonNode json) {
+        if (!json.has("signersPublicKeys") || !json.get("signersPublicKeys").isArray()) {
+            printGenerateCertProofInfoUsageMsg("wrong signersPublicKeys");
+            return;
+        }
+
+        List<String> signersPublicKeys = new ArrayList<String>();
+
+        Iterator<JsonNode> pksIterator = json.get("signersPublicKeys").elements();
+        while (pksIterator.hasNext()) {
+            JsonNode pkNode = pksIterator.next();
+
+            if (!pkNode.isTextual()) {
+                printGenerateCertProofInfoUsageMsg("wrong signersPublicKeys format");
+                return;
+            }
+
+            signersPublicKeys.add(pkNode.asText());
+        }
+
+        if (!json.has("mastersPublicKeys") || !json.get("mastersPublicKeys").isArray()) {
+            printGenerateCertProofInfoUsageMsg("wrong public keys");
+            return;
+        }
+
+        List<String> mastersPublicKeys = new ArrayList<String>();
```

Comment:
check that `mastersPublicKeys` size equals to `signersPublicKeys` and throw proper error.